pulsar-broker/src/main/java/org/apache/pulsar/broker/web/RestEndpointMetricsFilter.java

```diff
@@ -21,43 +21,68 @@
 import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
-import io.prometheus.client.Counter;
-import io.prometheus.client.Histogram;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
 import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
 import java.util.Stack;
+import javax.validation.constraints.NotNull;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
 import javax.ws.rs.container.ContainerResponseContext;
 import javax.ws.rs.container.ContainerResponseFilter;
 import javax.ws.rs.core.Response;
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.broker.stats.PulsarBrokerOpenTelemetry;
 import org.glassfish.jersey.server.internal.routing.UriRoutingContext;
 import org.glassfish.jersey.server.model.Resource;
 import org.glassfish.jersey.server.model.ResourceMethod;
-import org.jetbrains.annotations.NotNull;
 
 public class RestEndpointMetricsFilter implements ContainerResponseFilter, ContainerRequestFilter {
-    private static final LoadingCache<ResourceMethod, String> CACHE = CacheBuilder
+    private final LoadingCache<ResourceMethod, String> CACHE = CacheBuilder
             .newBuilder()
             .maximumSize(100)
+            .expireAfterAccess(Duration.ofMinutes(1))
             .build(new CacheLoader<>() {
                 @Override
                 public @NotNull String load(@NotNull ResourceMethod method) throws Exception {
                     return getRestPath(method);
                 }
             });
 
-    private static final Histogram LATENCY = Histogram
-            .build("pulsar_broker_rest_endpoint_latency", "-")
-            .unit("ms")
-            .labelNames("path", "method")
-            .buckets(10D, 20D, 50D, 100D, 200D, 500D, 1000D, 2000D)
-            .register();
-    private static final Counter FAILED = Counter
-            .build("pulsar_broker_rest_endpoint_failed", "-")
-            .labelNames("path", "method", "code")
-            .register();
-
     private static final String REQUEST_START_TIME = "requestStartTime";
+    private static final AttributeKey<String> PATH = AttributeKey.stringKey("path");
+    private static final AttributeKey<String> METHOD = AttributeKey.stringKey("method");
+    private static final AttributeKey<String> CODE = AttributeKey.stringKey("code");
+
+    private final DoubleHistogram latency;
+    private final LongCounter failed;
+
+    private RestEndpointMetricsFilter(PulsarService pulsar) {
+        PulsarBrokerOpenTelemetry telemetry = pulsar.getOpenTelemetry();
+        Meter meter = telemetry.getMeter();
+        latency = meter.histogramBuilder("pulsar_broker_rest_endpoint_latency")
```

Comment:
Please look at https://github.com/apache/pulsar/pull/22058 to understand how to do:
Instrument name, description, unit.